importlib_metadata/__init__.py

```diff
@@ -476,15 +479,46 @@ def zip_children(self):
         zip_path = zipp.Path(self.root)
         names = zip_path.root.namelist()
         self.joinpath = zip_path.joinpath
-
         return dict.fromkeys(child.split(posixpath.sep, 1)[0] for child in names)
 
-    def search(self, name):
-        return (
-            self.joinpath(child)
-            for child in self.children()
-            if name.matches(child, self.base)
-        )
+    def update_cache(self):
```

Comment:
That's because there's no more notion of "prepared computations" (because we just parse everything and store the normalized results into the cache).  "Matches" is basically just a dict lookup now (in `search()`).

I guess at best I could have a `def is_info(self, low): return low.endswith(...)` and `def is_egg(self, low): return self.base_is_egg and low == "egg-info"`.  I don't think adding these levels of indirection help legibility, but that's up to you.

I can't think of further factoring, though.